### PR TITLE
Moved connection UI to OpenRA.Mods.Common

### PR DESCRIPTION
**Pull request by Mailaender**
This is required for https://github.com/OpenRA/OpenRA/pull/4496.

### CODE REVIEW

File: OpenRA.Mods.Common/OpenRA.Mods.Common.csproj [hunk 1]
```diff
@@ -377,10 +385,23 @@
...
     <Compile Include="Widgets\Logic\ModBrowserLogic.cs" />
+    <Compile Include="Widgets\Logic\SettingsLogic.cs" />
+    <Compile Include="Widgets\Logic\TabCompletionLogic.cs" />
+    <Compile Include="Widgets\Logic\Networking\ClientTooltipLogic.cs" />
+    <Compile Include="Widgets\Logic\Networking\ConnectionLogic.cs" />
+    <Compile Include="Widgets\Logic\Networking\DirectConnectLogic.cs" />
+    <Compile Include="Widgets\Logic\Networking\DisconnectWatcherLogic.cs" />
+    <Compile Include="Widgets\Logic\Networking\KickClientLogic.cs" />
+    <Compile Include="Widgets\Logic\Networking\KickSpectatorsLogic.cs" />
+    <Compile Include="Widgets\Logic\Networking\LobbyLogic.cs" />
```
Comment (pchote) [inline OpenRA.Mods.Common/OpenRA.Mods.Common.csproj]: I'm not convinced that all of these really belong in Networking.  If you want to sub-group these, then `ClientTooltipLogic` and `LobbyLogic`, `LobbyMapPreviewLogic`, `LobbyUtils` would fit better in their own Lobby directory.

Comment (pchote) [inline OpenRA.Mods.Common/OpenRA.Mods.Common.csproj]: This also breaks the mapping between directories and namespaces.

**Comment by Mailaender** [comment]
Updated.

**Comment by Mailaender** [comment]
Rebased.

**Comment by obrakmann** [comment]
Compiles both with make and md, runs too. :+1: 

Needs a rebase, though.

**Comment by Mailaender** [comment]
Rebased.

**Comment by penev92** [comment]
:+1: after another rebase.

**Comment by Mailaender** [comment]
Rebased.

**Comment by reaperrr** [comment]
Needs another rebase...

**Comment by Mailaender** [comment]
Rebased again.

**Pull request merged**